packages/wallet/src/redux/sagas/blockchain-watcher.ts

```diff
@@ -0,0 +1,24 @@
+import { getProvider } from '../../utils/contract-utils';
+import { eventChannel } from 'redux-saga';
+import { take, put } from 'redux-saga/effects';
+import { blockMined } from '../actions';
+
+export function* blockchainWatcher() {
```

Comment:
@lalexgap Does this fire a redux action on every block mined while the app is running? If so, our redux state might get pretty noisy.

Reply:
@kerzhner Yup, were firing an action every time a block is mined which will get noisy. I'm not sure the best solution, we could try and be smarter about when we register the listener maybe?